xbmc/filesystem/FileCache.cpp

```diff
@@ -598,6 +598,7 @@ int CFileCache::IoControl(EIoControl request, void* param)
     status->currate = m_writeRateActual;
     status->lowrate = m_writeRateLowSpeed;
     m_writeRateLowSpeed = 0; // Reset low speed condition
+    m_bFilling = true;
```

Comment:
IMO this is the wrong fix. m_bFilling is to determine whether we started filling up from 0% to allow proper detection of low-speed conditions inside(!) CFileCache. Reading cache-status doesn't eg. reset cache so we shouldn't reset m_bFilling here either.

Like @FernetMenta says: probably the dependency from CVideoPlayer on CFileCache needs to go (if possible?). It has been there for ages and they never played nice. I tried fixing the "low-speed-logic" several times but to be honest I never completely understood the depending logic in CVideoPlayer...

please note the comment I once added in CFileCache:
   /* NOTE: We can only reliably test for low speed condition, when the cache is *really*
    * filling. This is because as soon as it's full the average-
    * rate will become approximately the current-rate which can flag false
    * low read-rate conditions.
    */

Reply:
I sincerely hope you are not suggesting users with low-speed internet should live with the video playback halting indefinitely until a "proper" fix is implemented.

In my humble opinion, this commit (nomatter it being the "wrong" fix) immediately relieves the user of ever having to face a stuck video situation which should be of a greater priority than leaving the users waiting for an extended period of time for an overhaul to the cache-videoplayer relationship.

I also hope you do not take me in the wrong way here, your explanation has been immensely helpful in making me understand why the code is the way it is, which only means my brain will keep hunting for the right answers! Haha

But as reasonable people here we ought to accept fixes for broken things while we keep doing what's best to make things better in the background without making end-users notice such obvious breakage.

Reply:
> I sincerely hope you are not suggesting users with low-speed internet should live with the video playback halting indefinitely until a "proper" fix is implemented.
> 
> In my humble opinion, this commit (nomatter it being the "wrong" fix) immediately relieves the user of ever having to face a stuck video situation which should be of a greater priority than leaving the users waiting for an extended period of time for an overhaul to the cache-videoplayer relationship.
> 
> I also hope you do not take me in the wrong way here, your explanation has been immensely helpful in making me understand why the code is the way it is, which only means my brain will keep hunting for the right answers! Haha
> 
> But as reasonable people here we ought to accept fixes for broken things while we keep doing what's best to make things better in the background without making end-users notice such obvious breakage.

I understand your frustrating but the proposed changes are simply not acceptable as it will break existing correct logic. So it *may* fix your issue but it may introduce new regressions we're not aware of. If you want to introduce a workaround/hack it must probably be done in VP, not CFileCache.